.github/workflows/static_checks.yml

```diff
@@ -151,39 +155,39 @@ jobs:
         shell: bash
 
       - name: Regex Patterns Validation Check
-        if: always()
+        if: ${{ !cancelled() }}
```

Comment:
We should probably have a comment for each of these lines explaining why it's needed (that it's basically a cancellation-compliant alternative to "if always()" per your findings).

Reply:
Sure, added the comment - 
```
# The expression if: ${{ !cancelled() }} runs a job or step regardless of its success or failure while responding to cancellations, 
serving as a cancellation-compliant alternative to if: ${{ always() }} in concurrent workflows.
```
for most of the occurrences, while updating the comments that existed as:
```diff
-if: ${{ always() }} # IMPORTANT: Upload reports regardless of status
+if: ${{ !cancelled() }} # IMPORTANT: Upload reports regardless of success or failure status
```

Would these phrasing be clear enough or would it require additional reference to any discussions?

Reply:
These seem really good--thanks!